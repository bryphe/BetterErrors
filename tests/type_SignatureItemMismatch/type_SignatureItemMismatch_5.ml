(* Test wrong type all by itself *)
module type Sig = sig
  val hasWrongType: string -> unit
end

module MyModule : Sig = struct
  let hasWrongType s = 0
end

