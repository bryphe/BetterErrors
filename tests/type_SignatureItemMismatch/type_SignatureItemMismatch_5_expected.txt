  > [4mUnformatted Error Output:[0m
  > File "tests/type_SignatureItemMismatch/type_SignatureItemMismatch_5.ml", line 6, characters 24-59:
  > Error: Signature mismatch:
  >        Modules do not match:
  >          sig val hasWrongType : 'a -> int end
  >        is not included in
  >          Sig
  >        Values do not match:
  >          val hasWrongType : 'a -> int
  >        is not included in
  >          val hasWrongType : string -> unit
  >        File "tests/type_SignatureItemMismatch/type_SignatureItemMismatch_5.ml", line 3, characters 2-34:
  >          Expected declaration
  >        File "tests/type_SignatureItemMismatch/type_SignatureItemMismatch_5.ml", line 7, characters 6-18:
  >          Actual declaration

[1m[31mError:[0m [4m[36mtests/type_SignatureItemMismatch/type_SignatureItemMismatch_5.ml:6:24-8:3[0m [1m[31m[0m
3 Â¦   val hasWrongType: string -> unit
4 Â¦ end
5 Â¦ 
6 Â¦ module MyModule : Sig = [4m[1m[31mstruct[0m
7 Â¦ [4m[1m[31m  let hasWrongType s = 0[0m
8 Â¦ [4m[1m[31mend[0m
9 Â¦ 

[1mThis module doesn't match its signature because the[0m [1mvalue[0m [1m[31mhasWrongType[0m [1mhas the wrong type[0m

  [2m[36mAt[0m [2m[36mtests/type_SignatureItemMismatch/type_SignatureItemMismatch_5.ml[0m[2m:3[0m
  The signatures says that [1mhasWrongType[0m should have type:

  [1m[32mhasWrongType:[0m
  [1m[32m'a => int[0m


  [2m[36mAt[0m [2m[36mtests/type_SignatureItemMismatch/type_SignatureItemMismatch_5.ml[0m[2m:7[0m
  But your module's [1mhasWrongType[0m has type:

  [1m[31mhasWrongType:[0m
  [1m[31mstring => unit[0m




